tests/api_jobs/test_downloads.py

```diff
@@ -505,27 +505,55 @@ def test_timeout_length_of_file_downloads(mocker, homedir, session, session_make
+    haf_GB_file = factory.File(source=source, is_downloaded=None, is_decrypted=None, size=500000000)
+    GB_file = factory.File(source=source, is_downloaded=None, is_decrypted=None, size=1000000000)
     session.add(source)
-    session.add(small_file)
-    session.add(large_file)
+    session.add(zero_byte_file)
+    session.add(one_byte_file)
+    session.add(KB_file)
+    session.add(fifty_KB_file)
+    session.add(half_MB_file)
+    session.add(MB_file)
+    session.add(five_MB_file)
+    session.add(haf_GB_file)
+    session.add(GB_file)
     session.commit()
 
     gpg = GpgHelper(homedir, session_maker, is_qubes=False)
-
-    small_job = FileDownloadJob(
-        small_file.uuid,
-        os.path.join(homedir, 'data'),
-        gpg,
-    )
-    large_job = FileDownloadJob(
-        small_file.uuid,
-        os.path.join(homedir, 'data'),
-        gpg,
-    )
-
-    small_file_timeout = small_job._get_realistic_timeout(small_file.size)
-    large_file_timeout = large_job._get_realistic_timeout(large_file.size)
-
-    assert small_file_timeout < large_file_timeout
+    zero_byte_file_job = FileDownloadJob(zero_byte_file.uuid, os.path.join(homedir, 'data'), gpg)
+    one_byte_file_job = FileDownloadJob(one_byte_file.uuid, os.path.join(homedir, 'data'), gpg)
+    KB_file_job = FileDownloadJob(KB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    fifty_KB_file_job = FileDownloadJob(fifty_KB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    half_MB_file_job = FileDownloadJob(half_MB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    MB_file_job = FileDownloadJob(MB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    five_MB_file_job = FileDownloadJob(five_MB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    half_GB_file_job = FileDownloadJob(haf_GB_file.uuid, os.path.join(homedir, 'data'), gpg)
+    GB_file_job = FileDownloadJob(GB_file.uuid, os.path.join(homedir, 'data'), gpg)
+
+    zero_byte_file_timeout = zero_byte_file_job._get_realistic_timeout(zero_byte_file.size)
+    one_byte_file_timeout = one_byte_file_job._get_realistic_timeout(one_byte_file.size)
+    KB_file_timeout = KB_file_job._get_realistic_timeout(KB_file.size)
+    fifty_KB_file_timeout = fifty_KB_file_job._get_realistic_timeout(fifty_KB_file.size)
+    half_MB_file_timeout = half_MB_file_job._get_realistic_timeout(half_MB_file.size)
+    MB_file_timeout = MB_file_job._get_realistic_timeout(MB_file.size)
+    five_MB_file_timeout = five_MB_file_job._get_realistic_timeout(five_MB_file.size)
+    GB_file_timeout = GB_file_job._get_realistic_timeout(GB_file.size)
+    half_GB_file_timeout = half_GB_file_job._get_realistic_timeout(haf_GB_file.size)
+
+    # timeout = ceil((file_size/100000)*1.5)+25
+    assert zero_byte_file_timeout == 25
+    assert one_byte_file_timeout == 26
+    assert KB_file_timeout == 26
+    assert fifty_KB_file_timeout == 26
+    assert half_MB_file_timeout == 33
+    assert MB_file_timeout == 40
```

Comment:
I have tested a 1 MB file download with 40 seconds timeout and 5 MB file with 100 second timeout successfully in staging